src/mrinufft/operators/off_resonance.py

```diff
@@ -0,0 +1,330 @@
+    return cp.asarray(input)
+
+
+def _to_torch(input):
+    xp = get_array_module(input)
+
+    if xp.__name__ == "numpy":
+        return torch.from_numpy(input)
+    elif xp.__name__ == "cupy":
+        return torch.from_dlpack(input)
+    else:
+        return input
+
+
+# */
+
+
+class MRIFourierCorrected(FourierOperatorBase):
+    """Fourier Operator with B0 Inhomogeneities compensation.
+
+    This is a wrapper around the Fourier Operator to compensate for the
+    B0 inhomogeneities  in the  k-space.
+
+    Parameters
+    ----------
+    fourier_op: object of class FourierBase
+        the fourier operator to wrap
+    fieldmap : np.ndarray, optional
+        Rate map defined as ``fieldmap = R2*_map + 1j * B0_map``.
+        ``*_map`` and ``t`` should have reciprocal units.
+        If ``zmap`` is real, assume ``zmap = B0_map``.
+        Also supports Cupy arrays and Torch tensors.
+        Expected shape is ``(nz, ny, nx)``.
+    readout_time : np.ndarray or GPUarray, optional
+        Readout time in ``[s]`` of shape ``(nshots, npts)`` or ``(nshots * npts,)``.
+        Also supports Cupy arrays and Torch tensors.
+    n_time_segments : int, optional
+        Number of time segments. The default is ``6``.
+    n_bins : int | Sequence[int] optional
+        Number of histogram bins to use for ``(B0, T2*)``. The default is ``(40, 10)``
+        If it is a scalar, assume ``n_bins = (n_bins, 10)``.
+        For real fieldmap (B0 only), ``n_bins[1]`` is ignored.
+    mask : np.ndarray or GPUarray, optional
+        Boolean mask to avoid histogram of background values.
+        The default is ``None`` (use the whole map).
+        Also supports Cupy arrays and Torch tensors.
+    B : np.ndarray, optional
+        Temporal interpolator of shape ``(n_time_segments, len(t))``.
+    C : np.ndarray, optional
+        Off-resonance phase map at each time segment center of shape
+        ``(n_time_segments, *fieldmap.shape)``.
+    backend: str, optional
+        The backend to use for computations. Either 'cpu', 'gpu' or 'torch'.
+        The default is `cpu`.
+    """
+
+    def __init__(
+        self,
+        fourier_op,
+        fieldmap=None,
+        readout_time=None,
+        n_time_segments=6,
+        n_bins=(40, 10),
+        mask=None,
+        B=None,
+        C=None,
+        backend="cpu",
+    ):
+        if backend == "gpu" and not CUPY_AVAILABLE:
+            raise RuntimeError("Cupy is required for gpu computations.")
+        if backend == "torch":
+            self.xp = torch
+        if backend == "gpu":
+            self.xp = cp
+        elif backend == "cpu":
+            self.xp = np
+        else:
+            raise ValueError("Unsupported backend.")
+        self._fourier_op = fourier_op
+
+        self.n_coils = fourier_op.n_coils
+        self.shape = fourier_op.shape
+        self.smaps = fourier_op.smaps
+        self.autograd_available = fourier_op.autograd_available
+
+        if B is not None and C is not None:
```

Comment:
```suggestion
        if B and C:
```

Reply:
Thanks! I initially accepted, but then reverted because, since B and C are arrays / tensors when provided, this line would return:

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()